server/src/test/java/org/opensearch/index/ShardIndexingPressureConcurrentExecutionTests.java

```diff
@@ -269,7 +269,11 @@ public void testCoordinatingPrimaryThreadedUpdateToShardLimitsAndRejections() th
 
         nodeStats = shardIndexingPressure.stats();
         IndexingPressurePerShardStats shardStoreStats = shardIndexingPressure.shardStats().getIndexingPressureShardStats(shardId1);
-        assertNull(shardStoreStats);
+        if (rejectionCount.get() == NUM_THREADS) {
```

Comment:
Should add a comment here explaining why there is a conditional check. Something like

```
// If rejection count equals NUM_THREADS that means rejections happened
// until the last request, then we'll get shardStoreStats which was updated on
// the last request. In other cases, the shardStoreStats simply moves to the
// cold store and null is returned.
```

Reply:
Sure, l will add it.